File: Guarana/Guarana.cpp
#include "Guarana.h"
#include "../Settings.h"

Guarana::Guarana(World *world, int x, int y)
    : Plant(GUARANA_STRENGTH, GUARANA_SYMBOL, GUARANA_COLOR, world, x, y) {}

void Guarana::action() {
  Plant::action();
}

void Guarana::collision(Organism* organism) {
  organism->setStrength(organism->getStrength() + 3);
  this->getWorld()->addLog("%s POS:(%d, %d) added 3 strength to %s POS:(%d, %d)",
                           this->getType().c_str(),
                           this->getPosition().getX(),
                           this->getPosition().getY(),
                           organism->getType().c_str(),
                           organism->getPosition().getX(),
                           organism->getPosition().getY());
}

std::string Guarana::getType() const { return "Guarana"; }

Organism* Guarana::createChild(int x, int y, int parent1Id, int parent2Id) {
  return new Guarana(this->getWorld(), x, y);
}

File: src/Guarana/Guarana.java
package Guarana;

import Field.Field;
import Organism.Organism;
import Organism.OrganismData;
import Plant.Plant;
import Settings.Settings;
import World.World;

public class Guarana extends Plant {

    public Guarana(World world, Field field) {
        super(new OrganismData(world.getNextOrganismId(), Settings.GUARANA_INITIATIVE, Settings.GUARANA_STRENGTH, Settings.GUARANA_IMAGE, -1, -1), world, field);
    }

    @Override
    public void action() {
        super.action();
    }

    @Override
    public void collision(Organism attacker) {
        if (attacker.isAnimal()) {
            attacker.getData().setStrength(attacker.getData().getStrength() + 3);
            this.world.addLog(this.getType() + " increased " + attacker.getType() + " strength by 3");
        }
    }

    @Override
    public String getType() {
        return "Guarana";
    }

    @Override
    public void reproduce(Field field) {
        Organism guarana = new Guarana(this.world, field);
        this.world.addOrganism(guarana);
        this.world.addLog(this.getType() + " reproduced (" + field.getNumber() + ")");
    }

}
